### ISSUE TITLE
Incorrect `[no-any-return]` failure in strict mode

### ISSUE THREAD

**Issue by facelessuser** [python/mypy#21008]
**Bug Report**

When doing float operations, `mypy` gets confused and thinks the result is `any` when it is clearly a `float`.

**To Reproduce**

This can be reproduced by copying the below code directly here: https://mypy-play.net/?mypy=latest&python=3.14&flags=strict

```python
def test() -> float:
    return (1.0 ** 2.0 / 1.0 ** 2.0 + 1.0 * (1.0 ** 2.0 + 1.0 ** 2.0) / 1.0 ** 2) ** (1.0 / 2.0)


test()
```

Result

```
main.py:2: error: Returning Any from function declared to return "float"  [no-any-return]
Found 1 error in 1 file (checked 1 source file)
```

**Expected Behavior**

This should not fail.

**Actual Behavior**

It fails.

**Your Environment**

- Mypy version used: 1.19.1
- Mypy command-line flags: strict
- Mypy configuration options from `mypy.ini` (and other config files):
- Python version used: Any

**Comment by facelessuser** [comment]
A simpler case. It seems to be related to `**` operator.

```py
def test() -> float:
    return 1.0 ** 2.0


test()
```

**Comment by hauntsaninja** [comment]
Yeah, the power operation between floats returns Any in typeshed. This is because it could return a complex, depending on the signs of the operands. Typing it as such would cause an unacceptable level of false positives
